plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -54,21 +72,16 @@
    * operation and display a general message that login is required but was cancelled or failed.
    *
    * Must be called from a UI context.
-   *
-   * @param shellProvider provides a shell for the login screen if login is necessary
-   * @throws IOException can be thrown by the underlying Login API request (e.g., network
-   *     error from the transport layer while sending/receiving a HTTP request/response.)
    */
-  public Credential getActiveCredential(IShellProvider shellProvider) throws IOException {
-    Credential credential = getCachedActiveCredential();
-
-    if (credential == null) {
-      credential = logIn(shellProvider);
-
-      IEclipseContext eclipseContext = PlatformUI.getWorkbench().getService(IEclipseContext.class);
-      eclipseContext.set(STASH_OAUTH_CRED_KEY, credential);
+  public Credential getActiveCredential() {
+    // TODO: holding a lock for a long period of time (especially when waiting for UI events)
```

Comment:
Good idea, but I think that'd be of no use and in fact not necessary, because this method is called only from the UI-thread. The UI-thread will never be blocked. (I made it that other methods that can be called from non-UI threads and holds a conflicting lock are always terminating quickly.) It's just that, if the UI-thread is holding a lock for a long time for taking care of login, other headless-jobs are blocked. Ideally, I think we should fix the ide-login library itself.